### PR TITLE
Changes in CI/CD Process

### LINKED ISSUE #40: Updating Deployment
**Describe the bug**
- With the new instance to deploy our project setup, deployment procedure needed to be changed.
- New setup uses systemD setup which needed to be managed by the deployment itself.

**Expected behavior**
- After the new changes, after building the jar file should be copied into the host deploy it.

### PR DESCRIPTION
**Pull request by picksitquick**
Fixes #40 

## Proposed Changes

  - we are now using password instead of SSH_Key, please make sure that the secrets are updated accordingly
  - systemD should now work with the deployment.
  - Need to test it out to know the exact behavior,. and fix any error that comes up.

### CODE REVIEW

File: deploy.sh [hunk 1]
```diff
@@ -1,42 +1,36 @@
...
 
-# Remove the temporary SSH key file
-rm /tmp/gamedoora-dev.pem
+# copy the new JAR file to the remote directory and rename it
+scp -i "/tmp/gamedoora-dev.pem" -o StrictHostKeyChecking=no $JAR_FILE "$SSH_USERNAME:$SSH_PASS@$SSH_HOST:$REMOTE_DIRECTORY/$REMOTE_JAR_FILE"
+ssh $SSH_USERNAME:$SSH_PASS@host 'sudo chown gamedoora:gamedoora /opt/gamedoora/\*.jar; sudo systemctl restart gamedoora-user-services'
 
 # Check the exit status of the previous command and set an output variable accordingly
-if [ $? -eq 0 ]; then
+if [ $? -eq 143 ]; then # changed success status
```
[the remark this answers is not in the document]
Comment (picksitquick) [reply deploy.sh:33]: SystemD file has Exit Success Status code as 143, so I thought after successfully executing the code I would get 143 as response, hence updated the success status code, is this the case?

[the remark this answers is not in the document]
Comment (aprajshekhar) [reply deploy.sh:33]: 143 is error status. So check for 0 as status code. Sorry forgot to mention that.

File: deploy.sh [hunk 2]
```diff
@@ -1,42 +1,36 @@
...
-echo "$SSH_KEY" > /tmp/gamedoora-dev.pem
-chmod 600 /tmp/gamedoora-dev.pem
-cat /tmp/gamedoora-dev.pem
 echo $SSH_USERNAME
 echo $SSH_HOST
 
-# connect to the EC2 instance and set up the remote directory
-ssh -i "/tmp/gamedoora-dev.pem" -o StrictHostKeyChecking=no "$SSH_USERNAME@$SSH_HOST" "
+# connect to the Server instance and set up the remote directory
+ssh -i "/tmp/gamedoora-dev.pem" -o StrictHostKeyChecking=no "$SSH_USERNAME:$SSH_PASS@$SSH_HOST" "
```
[the remark this answers is not in the document]
Comment (aprajshekhar) [reply deploy.sh]: Also, the password variable is SSH_PASSWORD

File: deploy.sh [hunk 3]
```diff
@@ -1,42 +1,36 @@
...
-  cd $REMOTE_DIRECTORY
-  chmod +x $REMOTE_JAR_FILE
-  sudo java -jar -Dserver.port=8081 $REMOTE_JAR_FILE > user_service.log 2> user_service.error.log &"
+# Stop the service
+ssh "$SSH_USERNAME:$SSH_PASS@$SSH_HOST" 'sudo systemctl daemon-reload; sudo systemctl stop gamedoora-user-services'
 
-# Remove the temporary SSH key file
-rm /tmp/gamedoora-dev.pem
+# copy the new JAR file to the remote directory and rename it
+scp -i "/tmp/gamedoora-dev.pem" -o StrictHostKeyChecking=no $JAR_FILE "$SSH_USERNAME:$SSH_PASS@$SSH_HOST:$REMOTE_DIRECTORY/$REMOTE_JAR_FILE"
```
[the remark this answers is not in the document]
Comment (aprajshekhar) [reply deploy.sh]: Same as my previous comment for password

**Comment by picksitquick** [comment]
As requested, changes have been made. keeping success code as 0 and checking for the same, also update the env variable name from SSH_PASS to SSH_PASSWORD.

**Comment by picksitquick** [comment]
We don't want to publish every code that gets updated from a PR, so in order to keep the deployment process separate, now on raising the PR only build will take place, and only when the PR is merged it will be deployed.
Future Scope: 
- Currently working on updating the build version directly from the pom.xml version and updating pom as well after every successful build and deploy, like a snapshot.

**Comment by picksitquick** [comment]
@aprajshekhar
The changes for separate workflow has been reverted, this branch can now be merged

**Review by aprajshekhar — APPROVED** [review]
LGTM.

**Pull request merged**